src/plutus/cek.md

```diff
@@ -0,0 +1,136 @@
+# The CEK Machine
+
+This page presents the operational semantics of the Plutus language using an abstract machine - a [CEK machine](https://en.wikipedia.org/wiki/CEK_Machine).
+Although more complex than traditional reduction semantics, the CEK machine offers an efficient evaluation strategy that enables precise modeling of resource usage and cost.
+The production Haskell evaluator is based on the CEK machine, and it also provides a practical foundation for alternative implementations.
+
+The following listing defines some key concepts of the CEK machine.
+
+```text
+Σ ∈ State ::= 𝑠; 𝜌 ⊳ 𝑀  // Computing M under environment 𝜌 with stack 𝑠
+            | 𝑠 ⊲ 𝑉     // Returning a value 𝑉 to stack 𝑠
+            | ⬥        // Throwing an error
+            | ◻ 𝑉      // Final state with result 𝑉
+
+𝑠 ∈ Stack ::= 𝑓*  // A stack has zero or more stack frames
+
+𝑉 ∈ CEK value ::= 〈con T 𝑐〉         // A constant 𝑐 with type T
+                | 〈delay 𝑀 𝜌〉      // A delayed computation, with an
+                                   // associated environment
+                | 〈lam 𝑥 𝑀 𝜌〉      // A lambda abstraction, with an
+                                   // associated environment
+                | 〈constr 𝑖 𝑉*〉   // A constructor application, where
+                                   // all arguments are values
+                | 〈builtin 𝑏 𝑉* 𝜂〉 // A builtin application with all supplied
+                                   // arguments as values, and expecting
+                                   // at least one more argument
+
+𝜌 ∈ Environment ::= []        // An empty environment
+                  | 𝜌[𝑥 ↦ 𝑉]  // Associate 𝑥 with 𝑉 in the environment
+
+𝜂 ∈ Expected builtin arguments ::= [𝜄]  // One argument
+                                 | 𝜄⋅𝜂  // Two or more arguments
+
+𝑓 ∈ Frame ::= (force _)    // Awaiting a delayed computation to be forced
+            | [_ (𝑀, 𝜌)]  // An application awaiting the function, where the
+                          // argument is a term associated with an environment
+            | [_ 𝑉]       // An application awaiting the function, where the
+                          // argument is a value
+            | [𝑉 _]       // An application awaiting the argument, where the
+                          // function is a value
+            | (constr 𝑖 𝑉* _ (𝑀*, 𝜌))  // A constructor application awaiting
+                                       // an argument. The arguments before
```

Comment:
Do you mean the arguments before `_`?  When we get one of these the Vs are arguments that we've already evaluated and the Ms are ones that we still have to evaluate, so "before" might also mean earlier in time.

Reply:
Updated to say "before the hole"